src/braket/pennylane_plugin/braket_device.py

```diff
@@ -605,6 +606,61 @@ def execute_and_gradients(self, circuits, **kwargs):
         res = res[0] if len(res) == 1 and active_jac else res
         return res, jacs
 
+    def _is_single_qubit_01_frame(self, f):
+        if self._device.arn == "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy":
+            return "drive" in f
+        else:
+            raise RuntimeError(
+                f"Single-qubit drive frame for pulse control not defined for "
+                f"device {self._device.arn}"
+            )
+
+    def _is_single_qubit_12_frame(self, f):
+        if self._device.arn == "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy":
+            return "second_state" in f
+        else:
+            raise RuntimeError(
+                f"Single-qubit drive frame for pulse control not defined for "
+                f"device {self._device.arn}"
+            )
+
+    def _get_frames(self, filter):
+        return {
+            f: info
+            for f, info in self._device.properties.pulse.dict()["frames"].items()
```

Comment:
Hope I am not missing something, just so you know the `device.frames` carries the same information (is a little bit shorter :))

(i.e. you can retrieve `device.frames["some_frame"].frequency`)

Reply:
Ah, thanks! I missed that 😆

Reply:
Actually, rolling this back and leaving as is because changing it made the tests more complicated; not worth the hassel 😅

Reply:
I'm curious why it broke. At first glance that line should be equivalent to 
```suggestion
            for f, info in self._device.frames.items()
```